docs/en/developers/04_Frontend/02_Design.md

```diff
@@ -1,11 +1,36 @@
-# Template file
+# Writing a new theme
 
-**TODO**
+**Note: Currently personal themes are not officially supported and may be overwritten when updating. Be sure to keep backups!**
 
-# Writing a new theme
+**As of writing (02-02-2021), support for themes as extensions is under development.**
+
+The easiest way to create a theme is by copying and modifying the base theme (or another of the pre-installed themes). The base theme can be found in [/p/themes/base-theme](https://github.com/FreshRSS/FreshRSS/tree/master/p/themes/base-theme). Themes require:
+- a **metadata.json** file to describe the theme.
+- a **loader.gif** file to use as a loading icon (assuming .loading's background has not been overridden)
+- an **icons** folder containing .svg, .ico, and .png files to override existing icons
+- a **thumbs** folder containing a file, **original.png** that will be used as the preview for the theme
+
+"_template.css" is normally added to the metadata file as a fallback for missing aspects. The file is taken from the base theme. If submitting a pull request for a theme, please know that [pull request themes must include this file.](https://github.com/FreshRSS/FreshRSS/pull/2938#issuecomment-624085450)
 
-**TODO**
+## RTL Support
+
+RTL (right-to-left) support for languages such as Hebrew and Arabic is handled automatically through CSSJanus. There should be no extra work required when developing the theme.
```

Comment:
That's not completely true in the current configuration. You do need to run CSSJanus (`make rtl`) and commit the result.